Source/WebKit/UIProcess/WebPageProxy.cpp

```diff
@@ -444,9 +454,28 @@ Ref<WebPageProxy> WebPageProxy::create(PageClient& pageClient, WebProcessProxy&
     return adoptRef(*new WebPageProxy(pageClient, process, WTFMove(configuration)));
 }
 
+WebPageProxy::Internals::Internals(WebPageProxy& page)
+    : page(page)
+    , audibleActivityTimer(RunLoop::main(), &page, &WebPageProxy::clearAudibleActivity)
+    , geolocationPermissionRequestManager(page)
+    , identifier(Identifier::generate())
+    , notificationManagerMessageHandler(page)
+    , pageLoadState(page)
+    , resetRecentCrashCountTimer(RunLoop::main(), &page, &WebPageProxy::resetRecentCrashCount)
+    , tryCloseTimeoutTimer(RunLoop::main(), &page, &WebPageProxy::tryCloseTimedOut)
+    , updateReportedMediaCaptureStateTimer(RunLoop::main(), &page, &WebPageProxy::updateReportedMediaCaptureState)
+    , webPageID(PageIdentifier::generate())
+#if HAVE(CVDISPLAYLINK)
+    , wheelEventActivityHysteresis([&page](PAL::HysteresisState state) { page.wheelEventHysteresisUpdated(state); })
+#endif
+#if ENABLE(VIDEO_PRESENTATION_MODE)
+    , fullscreenVideoTextRecognitionTimer(RunLoop::main(), &page, &WebPageProxy::fullscreenVideoTextRecognitionTimerFired)
+#endif
+{
+}
```

Comment:
WebPageProxy is *massive* (10k lines or so). As a separate step I think it would be great to move the Internals code into its own file, just to make it easier to navigate.

Reply:
The reason this file is so huge is that we accidentally created a [god object](https://en.wikipedia.org/wiki/God_object). I would be happy to separate this into separate files, or break it down. `WebCore::Frame` used to be like this but we worked hard to break it up. Internals itself is pretty small so I don’t think that helps much, and I think it’s good to have it next to the WebPageProxy constructor since both are initializing data members.